### ISSUE TITLE
Unable to play YouTube videos longer than four hours

### ISSUE THREAD

**Issue by jFalner** [anxdpanic/plugin.video.youtube#17]
Issue has existed through multiple versions including 6.8.4, and has been observed on Krypton and Leia versions.  Videos on YouTube exceeding four hours possibly might not play.  The issue is independent of whether proxy or MPEG-DASH is used or not.

Generally videos of four hours or less play successfully.  Videos of a length between four and six hours usually play, but sometimes fail.  Above six hours the failure rate becomes significant, and videos above seven hours are highly unlikely to play.

When failing, the following is observed:

• The screen dims and the blue circular Kodi spinner is very briefly seen (normal for any video playback).
• The screen regains full brightness, and a play prompt appears in the upper-right corner near the time (normal for any video playback).  The prompt consists of a blue play icon and the title of the video in truncated form.
• The screen again dims and the Kodi spinner reappears.
• After a few seconds (averaging eight to fifteen seconds), Kodi abruptly drops the spinner and returns the screen to full brightness.  The play prompt disappears with what appears to be a quick flash of the truncated portion of the title.

The only common denominator is length of video.  I've tried numerous videos from different channels and content creators, and it applies universally to all YouTube video.  Failure of long videos to play does not seem to affect the stability of the addon or Kodi; another video of an acceptable length can immediately be selected and played after a failure.

Environment is Kodi Leia 18.8 on Sony Bravia running Android Oreo 8.0.0.

**Comment by louhy** [comment]
This probably won't help with the OP's case but I do _not_ see this behavior with Kodi Leia 18.9 on Raspberry Pi 4 with LibreELEC 9.2.6 Linux 4.19.127; running v6.8.10 of this plugin. I routinely play 8hr+ long videos all the way through to the end. 

This is an example of a long video I often play if you ever try troubleshooting again. https://www.youtube.com/watch?v=QS_soXWWFrw

**Comment by jFalner** [comment]
Sorry for leaving this issue unattended for so long—I was preparing plenty of detailed testing and super-clean Kodi logs for troubleshooting, but the whole [YouTube buffering issue](https://github.com/anxdpanic/plugin.video.youtube/issues/163) came up, and suddenly my issue became _quite_ insignificant in comparison.

I will resume this when a stable release gets that other issue out of the way (just now noticed that new alpha3—fingers crossed!), but can report something interesting.  While waiting for our beloved @anxdpanic to work his magic, I implemented probonopd's [workaround involving the Invidious plugin](https://github.com/anxdpanic/plugin.video.youtube/issues/211#issuecomment-939445166), and to my surprise, the length issue is _gone_.  I not only successfully launched [this video](https://www.youtube.com/watch?v=vDwQpTFW2us), but it actually played to completion with nary a problem.  (Waking up to Mario still shouting "Wahoo!" was grating, but knowing the video still chugging away was lovely!  😁)

I've tried several other videos at well over twice the length of anything I have ever been able to play before, and they've all played just fine.  (I think a got a ten-hour video to play once, but that was a rarity.)  This eliminates any question about my RAM or device capabilities, I would think.  So once we have a stable release which addresses YouTube's new antisocial behavior, I'll return to this issue and do what I can to help troubleshoot.

Thanks all, particularly anxdpanic for the hard work!
